core/templates/domain/exploration/exploration-id-validation.service.ts

```diff
@@ -16,19 +16,31 @@
  * @fileoverview Service to validate exploration id
  */
 
-require('domain/summary/exploration-summary-backend-api.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationIdValidationService', [
-  'ExplorationSummaryBackendApiService',
-  function(ExplorationSummaryBackendApiService) {
-    return {
-      isExpPublished: function(explorationId) {
-        return ExplorationSummaryBackendApiService.
-          loadPublicExplorationSummaries([explorationId]).then(
-            function(summaries) {
-              return (summaries.length === 1 && summaries[0] !== null);
-            });
-      }
-    };
+import { ExplorationSummaryBackendApiService } from
+  'domain/summary/exploration-summary-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class ExplorationIdValidationService {
```

Comment:
No newline in between.
```suggestion
})
export class ExplorationIdValidationService {
```

Reply:
done!